packages/cli/src/models/network/NetworkController.ts

```diff
@@ -671,6 +690,35 @@ export default class NetworkController {
     }
   }
 
+  public async createInstance(packageName: string, contractAlias: string, initArgs: unknown[]): Promise<Contract> {
+    await this.migrateManifestVersionIfNeeded();
+
+    if (!packageName) {
+      packageName = this.projectFile.name;
+    }
+
+    // TODO: Fix for contracts from dependencies that may use libraries.
+    if (packageName === this.projectFile.name) {
+      await this.deployChangedSolidityLibs(contractAlias);
+    }
+
+    const contract = this.contractManager.getContractClass(packageName, contractAlias);
+    await this._setSolidityLibs(contract);
+
+    const instance = await Transactions.deployContract(contract, initArgs, this.txParams);
+
+    if (packageName === this.projectFile.name) {
+      await this._updateTruffleDeployedInformation(contractAlias, instance);
+    }
+
+    this.networkFile.addProxy(packageName, contractAlias, {
```

Comment:
WDYT about storing the contract bytecode hash as well? It may come in handy in the future if we want to tell the user whether a deployed instance matches the current source code or not.